pkg/helm/utils.go

```diff
@@ -124,11 +128,63 @@ func generateRandomName(seed *int64) (string, error) {
 	return fmt.Sprintf("%s-%s", adjective, animal), nil
 }
 
+func persistHostname(hostname string) error {
+	filePath := local.GetInstallationHostnamePath()
+	err := os.WriteFile(filePath, []byte(hostname), 0644)
+
+	if err != nil {
+		return fmt.Errorf("error persisting hostname: %v", err)
+	}
+	return nil
+}
+
+// Read hostname from /var/lib/tensorleap or return empty if file does not exist
+func readHostname() (string, error) {
+	filePath := local.GetInstallationHostnamePath()
+	data, err := os.ReadFile(filePath) // Reading the content of the file
+
+	if err != nil {
+		if os.IsNotExist(err) {
+			return "", nil // Returning an empty string and nil error if the file does not exist
+		}
+		return "", fmt.Errorf("error reading hostname: %v", err) // Returning the error if other error occurs
+	}
+
+	return string(data), nil
+}
+
 func readOrGenerateHostname() (string, error) {
-	return "TBD", nil
+	existingHostName, err := readHostname()
```

Comment:
Consider using the computer username and some random suffix that will be created on installation (save it on a file in the manifest directory)- it seems to me as a better indication of who the computer belongs

Reply:
I thought about it, and there are some downsides to it. e.g. you'll get a whole bunch of `ubuntu` or `root` hostnames. But it's possible to also do both - a unique generated name + something identifiable. Ultimately, though, none of these are reliably helpful. What we need is a way to tie some actual customer identifier to the installation, and this should be done in some provisioning step where we either ask the user to identify himself or perform some validation when we generate the datadog API keys for that user.
The random name approach is just meant to be statistically (pretty) unique.

Reply:
asking the user for an identifier sounds good. we can use the domain or check the user name and if it is not root or ubuntu we can use it + some short generated suffix. otherwise using any other random name.

Reply:
sounds good